internal/heartbeat/heartbeat.go

```diff
@@ -85,6 +90,42 @@ func (s *HeartbeatData) RetrieveInfo() models.Heartbeat {
 	return heartbeatInfo
 }
 
+func (s *HeartbeatData) buildHardwareInfo() *models.HardwareInfo {
+	var hardwareInfo *models.HardwareInfo
+	var err error
+	if s.previousMutableHardwareInfo == nil {
+		// Only send all Hw info (mutable + immutable) for the 1st heartbeat, then send only mutable hw info
+		hardwareInfo, err = s.hardware.GetHardwareInformation()
+		if err != nil {
+			log.Errorf("cannot get full hardware information. DeviceID: %s; err: %v", s.workloadManager.GetDeviceID(), err)
+		}
+		s.previousMutableHardwareInfo = s.hardware.CreateHardwareMutableInformation()
+		if err != nil {
+			log.Errorf("cannot get mutable hardware information. DeviceID: %s; err: %v", s.workloadManager.GetDeviceID(), err)
+		}
```

Comment:
as a rule of dumb, I will do a `return hardwareInfo` here, and remove the else, it's a bit clear what it's doing ;-)

Reply:
good idea for clarity